dataTypes/dataTypes.go

```diff
@@ -14,24 +15,40 @@ type Field struct {
 
 type FieldMap struct {
 	Candidate Candidate
-}
-
-type TransformedData struct {
-	Candidate [2]string
+	Job Job
 }
 
 type ElectiveResponse struct {
-	TransformedData CandidateResponse
+	TransformedData interface{}
 	entityType string
 	error string
 }
 
-func ToElectiveStruct(fieldMap FieldMap, data map[string]string) ElectiveResponse{
-	rep := ElectiveResponse{}
+func ToElectiveStruct(fieldMap []byte, data []byte) ElectiveResponse{
+	ret := ElectiveResponse{}
+
+	var fieldInfo FieldMap
+	err := json.Unmarshal(fieldMap, &fieldInfo)
+	if err != nil {
+		panic(err)
+	}
+
+	var dataInfo map[string] string
+	err = json.Unmarshal(data, &dataInfo)
+	if err != nil {
+		panic(err)
```

Comment:
To quote Alex "Panic is a sackable offence ..." That will terminate program (service). Better to log and (or) return error